### PR TITLE
feat(backends): support globbing in existing file-reading backends

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds support for reading globs to existing file-supporting backends with a few exceptions. The clickhouse backend does not support reading JSON (no way to infer the schema), and I simply did not implement datafusion json reading.

### CODE REVIEW

File: ibis/backends/clickhouse/__init__.py [hunk 1]
```diff
@@ -548,22 +549,28 @@ def read_parquet(
         self,
         path: str | Path,
         table_name: str | None = None,
-        engine: str = "File(Parquet)",
+        engine: str = "File(Native)",
```
Comment (cpcloud) [inline ibis/backends/clickhouse/__init__.py:552]: This had to be changed to allow appending multiple files to the same table. `File(Parquet)` doesn't support this without a server side setting, which may not be configurable by the user.

**Comment by cpcloud** [comment]
Cloud backend tests are passing:

```
…/ibis on  read-globs is 📦 v6.1.0 via 🐍 v3.10.12 via ❄️  impure (ibis-3.10.12-env) took 9s
❯ pytest -m 'bigquery or snowflake' -n auto --dist=loadgroup --snapshot-update -q
bringing up nodes...
................................x...................x...x.......x.....x......x...............x.x....x...x....x......x..xx...x...xx..................x...xx..s......x.s..x.x..x..x...s...x..x.. [  7%]
...x...x......x...x..........x..x..x..............x..................xx........x...x.x.........x.xx.xx..............x......x...................x.............................................. [ 14%]
...............................................................x...........x....................................x.....................x...x.xx........x.x.xx.xx.xxxx.xx.xx.xxxxxxxx....xxxxxx. [ 21%]
x..xx.......xx.xxx.x.....x.x...x.........xx...x....x.....x.x.......xx.xxssss............x..x......x..x...............x..x......................x...............x............................xx [ 28%]
....x...xx.xx....x...........x.x..xxxx.......x..x..xx.....x.......s....x...x.......x...xx.x...x...xxxxxx.xx.xxxxxxxxx.......xxx..x.xxx.xx..xx.x.x..x...x.x.xx.....x...x........x....xx....x.xx [ 35%]
x.....x.x...x.x.xx.x.....xx...xxx..x.x..xxx..x.x....xx...x.x.xx.x..x....x..x...x..x.....x..x.x..x....xxx....x...x..x..........xxxx.xxxxx.....xx.............x.....xx..xx............x...x..... [ 42%]
x......xx..x...x...x.x......x...........xx..x....xx...x.............x.x.........x..............x...xx.......x.....x..xx....x....x.....x....xx.....x...xxxx.x.xx..x.....xx..xx.....x..x...x.... [ 49%]
.x......x.....xxx...x.....x.......xx..xx.....x.x.x...x...xx.xx................................................................x......x....................x..........x.............x.......... [ 56%]
...........xx...........x....x.......x......x..........................................x......x.x.xx....x.......xx.x.....xx.xxx.x.........x.xx...xx.xx.....x.....x...............x........x... [ 63%]
.x...............x......x......s....s.........s........x...s..x.x.........x...............x..x...........x..xxx......x....x........x.....xxx..x.xx......x......x..........x.x...........x..... [ 70%]
.....x..x...x...........xxs..x.............s...x.......................................x..............x.....xx.x.x..............x.......x...x..........x............x......................... [ 78%]
.........x....x............................xx............x..xx................................................................................................................................ [ 85%]
..................................................................s...................................................................x.x....x..x......xx......xx...........x......xx......... [ 92%]
..................................xx...........................................................................................s..........................xx.x...x..x.xx...................... [ 99%]
...................                                                                                                                                                                            [100%]
2248 passed, 16 skipped, 415 xfailed in 366.31s (0:06:06)
```

**Review by gforsyth — COMMENTED** [review]
Implementation looks good!  I flagged two spots where docstrings should be updated -- I think there's a third in the snowflake `read_csv` but github doesn't believe in commenting outside of diffs.

we should also, i think, standardize on whether we want to think of providing a path to a directory of files as a supported operation that is different from a glob, or whether it's an implicit glob, and update docs accordingly, but that can happen in a followup.

File: ibis/backends/snowflake/__init__.py [hunk 2]
```diff
@@ -774,7 +789,7 @@ def read_parquet(
             Path to a Parquet file
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/snowflake/__init__.py]: Fixed

**Comment by cpcloud** [comment]
I'll rebase.

**Review by gforsyth — APPROVED** [review]
:shipit:

**Pull request merged**